src/common/types/types.cpp

```diff
@@ -149,6 +149,22 @@ std::unique_ptr<StructField> StructField::copy() const {
     return std::make_unique<StructField>(name, type->copy());
 }
 
+StructTypeInfo::StructTypeInfo(std::vector<std::unique_ptr<StructField>> fields)
+    : fields{std::move(fields)} {
+    for (auto i = 0u; i < this->fields.size(); i++) {
+        fieldNameToIdxMap.emplace(this->fields[i]->getName(), i);
+    }
+}
+
+struct_field_idx_t StructTypeInfo::getStructFieldIdx(std::string fieldName) const {
+    StringUtils::toUpper(fieldName);
+    if (fieldNameToIdxMap.contains(fieldName)) {
+        return fieldNameToIdxMap.at(fieldName);
+    }
+    throw common::BinderException{
```

Comment:
Don't return BinderException, this is a common function. You can return an invalid idx here, and add check logic in binder to throw BinderException.
Btw, I don't think you need to add a new alias type `struct_field_idx_t`. Just use `vector_idx_t`.